hail/python/hailtop/__init__.py

```diff
@@ -1,3 +1,13 @@
+import asyncio
+
+try:
+    asyncio.get_running_loop()
+    import nest_asyncio
+    nest_asyncio.apply()
+except RuntimeError as err:
+    assert err.args[0] == "no running event loop"
+    asyncio.set_event_loop(asyncio.new_event_loop())
```

Comment:
I'm still chasing down exactly why this is a problem but I think this creates an event loop in a thread that is not the Main Thread. Asyncio wants there to be one single event loop in the entire program, not one per thread, so we error later when a test, running in the Main Thread, calls `asyncio.get_event_loop()`. That function refuses to create a new loop if `asyncio.set_event_loop` was already called.

High level: why create the event loop here? If its not running, just do nothing, yeah?

Reply:
OK, I think I've made sense of this.

You *really* don't want to have two event loops. `asyncio.run` creates an event loop, runs the coro, then destroys the event loop. `aiohttp` (and perhaps other libraries) stash a reference to the event loop. They cannot be used across multiple calls to `asyncio.run`.

For this reason, asyncio works to prevent you from creating two event loops. Pytest appears to use multiple threads to `import` all the test modules. This change forces initialization of an event loop in some non-Main-Thread. That causes `self._local._set_called` to be set to `True` on the `_UnixDefaultEventLoopPolicy`. That prevents `get_event_loop` from creating new threads.

So we're stuck: the event loop doesn't exist in the main thread and we're not allowed to create a new one.